pkg/http/core/artifacts.go

```diff
@@ -212,10 +218,143 @@ func GetArtifact(c *gin.Context) {
 			b = []byte(response.Content)
 		}
 
+	case artifact.TypeGitRepo:
+		gc, err := cc.GitRepoClientForAccountName(a.ArtifactAccount)
+		if err != nil {
+			clouddriver.WriteError(c, http.StatusBadRequest, err)
+			return
+		}
+
+		branch := "master"
+		if a.Version != "" {
+			branch = a.Version
+		}
+
+		url := fmt.Sprintf("%s/archive/%s.tar.gz", a.Reference, branch)
```

Comment:
Is there a reason we went with `https://github.com/User/repo/archive/master.tar.gz` over the v3 API `https://api.github.com/repos/User/repo/tarball/master`?

https://stackoverflow.com/questions/8377081/github-api-download-zip-or-tarball-link

Reply:
No, just less URL manipulation from what the user enters in on the UI
![image](https://user-images.githubusercontent.com/16955707/99728178-08b58f00-2a87-11eb-87f9-1a5fb73356b3.png)

Which reminds me - I only added support for token auth, not ssh keys

Reply:
Oh right, that would be a pain to deal with. LGTM - I'll wait on ssh keys (and possibly a test in sandbox, if it's not done already) to merge.